src/layers/icmpv6.rs

```diff
@@ -31,6 +30,55 @@ pub const ICMPV6_NEIGHBOR_SOLICITATION: u8 = 135;
 pub const ICMPV6_NEIGHBOR_ADVERTISEMENT: u8 = 136;
 pub const ICMPV6_REDIRECT: u8 = 137;
 
+fn handle_icmpv6_options(bytes: &[u8], decoded: &mut usize) -> Vec<IcmpV6Option> {
```

Comment:
The signature of this function need not send `&mut u8` to the caller. Instead a more semantic way is the return the decoded bytes count from the function itself. So the return type could be  - `Result<(usize, Vec<IcmpV6Option), Error>`. The `decoded` can then be taken from the `Ok` value of the result.

Reply:
See below for the details of how you can send the current offset without having to send the `decoded` as a param,